src/neural/dx/MetaCommand.h

```diff
@@ -0,0 +1,128 @@
+/*
+  This file is part of Leela Chess Zero.
+  Copyright (C) 2020 The LCZero Authors
+
+  Leela Chess is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  Leela Chess is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with Leela Chess.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#pragma once
+
+// Obtained by using EnumerateMetaCommands and EnumerateMetaCommandParameters
+// calls. Simplified a bit.
+
+struct TensorDesc {
+  uint64_t DataType;
+  uint64_t Flags;
+  uint64_t DimensionCount;
+  uint64_t Size[5];
+  uint64_t Stride[5];
+  uint64_t StrideAlignment[5];
+  uint64_t BaseAlignmentInBytes;
+  uint64_t PhysicalSizeInElements;
+};
+
+//----------------------------------------------------------------------------------//
+// GEMM (Matrix multiply)
+//----------------------------------------------------------------------------------//
+
+const GUID GemmGuid = {0x1e52ebab,
+                       0x25ba,
+                       0x463a,
+                       {0xa2, 0x85, 0x0a, 0x78, 0x8e, 0xef, 0x5d, 0x01}};
+
+struct GemmCreateDesc {
+  TensorDesc DescA;
+  TensorDesc DescB;
+  TensorDesc DescC;
+  uint64_t cMatrixNull;
+  TensorDesc DescOut;
+
+  uint64_t Precision;
+  uint64_t TransA;
+  uint64_t TransB;
+  float Alpha;
+  float Beta;
+
+  uint64_t ActivationFunction;
+  float ActivationParam1, ActivationParam2;
+  uint64_t ActivationIsNull;
+  uint64_t BindFlags;
+};
+
+struct GemmInitDesc {
+  D3D12_GPU_DESCRIPTOR_HANDLE AResource;
+  D3D12_GPU_DESCRIPTOR_HANDLE BResource;
+  D3D12_GPU_DESCRIPTOR_HANDLE CResource;
+  D3D12_GPU_DESCRIPTOR_HANDLE PersistentResource;
+  D3D12_GPU_DESCRIPTOR_HANDLE TemporaryResource;
+};
+
+struct GemmExecuteDesc {
+  D3D12_GPU_DESCRIPTOR_HANDLE AResource;
+  D3D12_GPU_DESCRIPTOR_HANDLE BResource;
+  D3D12_GPU_DESCRIPTOR_HANDLE CResource;
+  D3D12_GPU_DESCRIPTOR_HANDLE OutputResource;
+
+  D3D12_GPU_DESCRIPTOR_HANDLE PersistentResource;
+  D3D12_GPU_DESCRIPTOR_HANDLE TemporaryResource;
+};
+
+//----------------------------------------------------------------------------------//
+// Convolution
+//----------------------------------------------------------------------------------//
+
+GUID ConvGuid = {0x17804d6b,
```

Comment:
`const`, otherwise it will have external linkage and will cause a linker error if multiple files will include this.

Reply:
Right. I missed adding const here. Changed to contexpr like the one above.